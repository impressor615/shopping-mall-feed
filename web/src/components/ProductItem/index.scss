.product-item {
  display: inline-block;
  position: relative;

  .product-desc,
  .product-price {
    position: absolute;
    bottom: 5%;
    left: 5%;

    color: white;
    font-weight: 500;
  }

  .product-desc {
    font-size: 28px;
  }

  .product-desc div:first-child {
    margin-bottom: 10px;
  }

  .product-price {
    font-size: 22px;
  }
}