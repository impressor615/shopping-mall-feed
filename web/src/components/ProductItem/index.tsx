import "./index.scss";

import React from "react";

interface ProductItemProps {
  url: string;
  imgSrc: string;
  price: number;
  name: string;
  shopName: string;
  big?: boolean;
}

interface ProductDescriptionProps {
  price: number;
  name: string;
  shopName: string;
  big?: boolean;
}

export const ProductDescription = ({ big, shopName, name, price }: ProductDescriptionProps) => (
  <>
    {
      big ? (
        <div className="product-desc">
          <div>{ `#${shopName}` }</div>
          <div>{ name }</div>
        </div>
      ) : <div className="product-price">{ `${price.toLocaleString()}Ïõê` }</div>
    }
  </>
)

const ProductItem = ({ url, imgSrc, price, name, shopName, big }: ProductItemProps) => (
  <a className="product-item" href={url} target="_blank">
    <img src={imgSrc} alt="product" />
    <ProductDescription
      price={price}
      big={big}
      name={name}
      shopName={shopName}
    />
  </a>
);

export default ProductItem;
