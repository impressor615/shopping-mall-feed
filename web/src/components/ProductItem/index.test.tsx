import enzyme from "enzyme";
import React from "react";

import ProductItem, { ProductDescription } from "@/components/ProductItem";

describe("ProductItem", () => {
  it("should render img and ProductionDescription", () => {
    const Component = (
      <ProductItem
        url="url"
        imgSrc="image"
        price={10000}
        name="product-name"
        shopName="shop-name"
      />
    );
    const wrapper = enzyme.shallow(Component);
    expect(wrapper.find(ProductDescription)).toHaveLength(1);
    expect(wrapper.find(".product-item")).toHaveLength(1);
    expect(wrapper.find("img")).toHaveLength(1);
  });

  it("should render product-desc when big is true", () => {
    const Component = (
      <ProductItem
        big
        url="url"
        imgSrc="image"
        price={10000}
        name="product-name"
        shopName="shop-name"
      />
    );

    const wrapper = enzyme.mount(Component);
    expect(wrapper.find(".product-desc")).toHaveLength(1);
  });

  it("should be matched with snapshot", () => {
    const Component = (
      <ProductItem
        url="url"
        imgSrc="image"
        price={10000}
        name="product-name"
        shopName="shop-name"
      />
    );
    const wrapper = enzyme.mount(Component);
    expect(wrapper).toMatchSnapshot();
  });
});
