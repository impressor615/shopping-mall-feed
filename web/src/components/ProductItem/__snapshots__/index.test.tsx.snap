// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem should be matched with snapshot 1`] = `
<ProductItem
  imgSrc="image"
  name="product-name"
  price={10000}
  shopName="shop-name"
  url="url"
>
  <a
    className="product-item"
    href="url"
    target="_blank"
  >
    <img
      alt="product"
      src="image"
    />
    <Component
      name="product-name"
      price={10000}
      shopName="shop-name"
    >
      <div
        className="product-price"
      >
        10,000Ïõê
      </div>
    </Component>
  </a>
</ProductItem>
`;
